code/controllers/subsystem/minimaps.dm

```diff
@@ -226,6 +226,7 @@ SUBSYSTEM_DEF(minimaps)
 		if(!(earlyadds["[target.z]"]))
 			earlyadds["[target.z]"] = list()
 		earlyadds["[target.z]"] += CALLBACK(src, PROC_REF(add_marker), target, hud_flags, blip)
+		RegisterSignal(target, COMSIG_QDELETING, PROC_REF(remove_earlyadd), override = TRUE) //if it was loaded before the z-level
```

Comment:
This never gets unregistered if its processed normally is that intentional

Reply:
Also you need to write a detailed explanation for using override = true

Reply:
when the zlevel finished loading, the signal gets overridden